core/src/main/scala/scodec/bits/BitVector.scala

```diff
@@ -1013,6 +1016,10 @@ sealed trait BitVector extends BitwiseOperations[BitVector, Long] with Serializa
    */
   final def digest(digest: MessageDigest): BitVector = BitVector(bytes.digest(digest))
 
+  final def cipher(ci: Cipher, key: SecretKeySpec, opmode: Int): BitVector = BitVector(bytes.cipher(ci, key, opmode))
+  final def encrypt(ci: Cipher, key: SecretKeySpec): BitVector = cipher(ci, key, Cipher.ENCRYPT_MODE)
```

Comment:
Can you change `SecretKeySpec` to `Key`? That will let this work with more algorithms, including asymmetric ones.